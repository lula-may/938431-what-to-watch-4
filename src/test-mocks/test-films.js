const movieTitles = [
  `Gone With The Wind`,
  `The Godfather`,
  `Casablanca`,
  `A Streetcar Named Desire`,
  `Some Likes It Hot`,
];

const movies = movieTitles.map((title, i) => {
  return {
    title,
    id: i,
    src: `poster-${i}.jpg`
  };
});

const headerMovie = {
  title: `Jurassic Park`,
  genre: `Sci-Fi`,
  releaseYear: 1993,
  bg: `background.jpg`,
  poster: `header-poster.jpg`
};

export {movies, headerMovie};

