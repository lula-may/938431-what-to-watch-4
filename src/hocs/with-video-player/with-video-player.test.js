import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withVideoPlayer from "./with-video-player.jsx";
import {testMovies} from "../../test-mocks/test-films.js";

const movie = testMovies[0];

const MockComponent = (props) => {
  const {children} = props;
  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.node.isRequired,
};

const MockComponentWrapped = withVideoPlayer(MockComponent);

describe(`WithVideoPlayer HOC`, () => {
  it(`should render <video/> inside MockComponent`, () => {
    const tree = renderer.create(
        <MockComponentWrapped
          movie={movie}
          onCardEnter={() => {}}
          onCardClick={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});


