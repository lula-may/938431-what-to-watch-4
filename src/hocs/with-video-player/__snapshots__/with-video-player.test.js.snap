// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithVideoPlayer HOC should render <video/> inside MockComponent 1`] = `
<div>
  <video
    height="175"
    poster="poster-0.jpg"
    width="280"
  />
</div>
`;
